k2/csrc/ragged_inl.h

```diff
@@ -95,6 +100,35 @@ Ragged<T> RandomRagged(T min_value, T max_value, int32_t min_num_axes,
   return Ragged<T>(shape, values);
 }
 
+template <typename T, typename Op /* = LessThan<T> */>
+void SortSublists(Ragged<T> *src, Array1<int32_t> *order) {
+  K2_DCHECK(IsCompatible(src->values, *order));
+  K2_DCHECK_EQ(src->values.Dim(), order->Dim());
+  K2_DCHECK_EQ(src->Context()->GetDeviceType(), kCuda)
+      << "It supports only CUDA at present";
+
+  std::unique_ptr<mgpu::context_t> context =
+      GetModernGpuAllocator(src->Context()->GetDeviceId());
+
+  Array1<int32_t> &segment = src->shape.RowSplits(src->NumAxes() - 1);
```

Comment:
it's better to add `K2_CHECK_EQ(src->shape.NumAxes() == 2` at entry as we just sort values in `src`

Reply:
It supports ragged arrays with more than two axes.

Reply:
Well, at least check `>=2` here otherwise there will be an error when calling `shape.RowSplits(NumAxes() - 1)`, note that you should add such requirements in the documentation https://github.com/k2-fsa/k2/blob/a17852434c87f76e2dfc837918197c98b7cb947d/k2/csrc/ragged.h#L577-L584



BTW, in other APIs, such as `MaxPerSublist`, we require `NumAxes() ==2` now. @danpovey I think we should make those api consistent,  should we require exactly `NumAxes() == 2` or `NumAxes() >=2`

Reply:
In general I'd like to make the APIs as general as they can be made without requiring extra work.

Reply:
It has no requirement about the number of axes a ragged array has. It always sorts the last axis.
This function works as long as the ragged array is not empty.

Reply:
mm.. definitely it requires NumAxes >=2...It's meaningless to call RowSplits(1) on a shape with `NumAxes <  2`, actually it will crash..

Then I suggest we add documentation and requirements for those APIs with `axes >= 2`

Reply:
Could you give an example to show what a ragged array with `NumAxes < 2` looks like?

Reply:
mm, the point here is, now we allow user define a empty RaggedShape technically, https://github.com/k2-fsa/k2/blob/a17852434c87f76e2dfc837918197c98b7cb947d/k2/csrc/ragged.h#L137, so it's better if we can check this and report error as early as possible (i.e. at entry of the function).

Reply:
I'll add a `K2_DCHECK` in the next commit.

Reply:
> it's better to add K2_CHECK_EQ(src->shape.NumAxes() == 2 at entry as we just sort values in src

@qindazhu 
fixed in #218 

https://github.com/k2-fsa/k2/blob/d291daf7a88d50907d2b1039ca6033f307995a65/k2/csrc/ragged_inl.h#L111